sparql-queries/sparql-queries.js

```diff
@@ -76,66 +99,183 @@ export async function finalizePublications(pubTask) {
-            
-            INSERT DATA {
-              GRAPH <http://mu.semte.ch/graphs/system/email> {
-                <http://themis.vlaanderen.be/id/emails/${uuid}> a nmo:Email;
-                    mu:uuid ${uuid} ;
-                    nmo:emailBcc ${batch};
-                    nmo:messageFrom ${EMAIL_FROM};
-                    nmo:messageSubject "Persbericht vlaamse overheid: ${pubTask.title.value}";
-                    nmo:htmlMessageContent ${sparqlEscapeString(html)};
-                    nmo:sentDate ${sparqlEscapeDateTime(now)};
-                    nmo:isPartOf ${sparqlEscapeUri(outbox)} .
-              }
-            }
-        `);
+                ${PREFIXES}
+
+                INSERT DATA {
+                  GRAPH <http://mu.semte.ch/graphs/system/email> {
+                    <http://themis.vlaanderen.be/id/emails/${uuid}> a nmo:Email;
+                        mu:uuid ${uuid} ;
+                        nmo:emailTo ${EMAIL_TO};
+                        nmo:emailBcc ${batch.toString()};
+                        nmo:messageFrom ${EMAIL_FROM};
+                        nmo:messageSubject "${pubTask.title}";
+                        nmo:htmlMessageContent ${sparqlEscapeString(html)};
+                        nmo:sentDate ${sparqlEscapeDateTime(now)};
+                        ${attachmentsQuery}
+                        nmo:isPartOf ${sparqlEscapeUri(outbox)} .
+                  }
+                }
+            `);
     }
 
-    // TODO: add attachments
-    // nmo:hasAttachment <http://mu.semte.ch/services/file-service/files/602fa6c6424d81000d000002> ;
-    // nmo:hasAttachment <http://mu.semte.ch/services/file-service/files/602fa6c6424d81000d000000> .
 }
 
-function createRecipientBatches(pubTask){
-    // TODO: create batches from recipient email addr
-    return [];
+export async function createRecipientBatches(pubTask) {
+
+    const q = await query(`
+     ${PREFIXES}
+     
+     SELECT ?contactListItemValue ?contactItemValue
+     WHERE {
+         GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+             {
+                  ?contactList      ext:contactListHasChannelPublicationEvent   ${sparqlEscapeUri(pubTask.pubEvent)};
+                                    nco:containsContact                         ?contactListItem.
+                  ?contactListItem  vcard:hasEmail                              ?email.
+                  ?email            vcard:hasValue                              ?contactListItemValue. 
+              } 
+              UNION 
+              {
+                  ?contactItem      ext:contactHasChannelPublicationEvent       ${sparqlEscapeUri(pubTask.pubEvent)};
+                                    vcard:hasEmail                              ?email.
+                  ?email            vcard:hasValue                              ?contactItemValue.
+              }
+         } 
+     }
+    `);
+
+    // create array of email values
+    const emails = q.results.bindings.map((item) => {
```

Comment:
Ik denk dat je dit kan vermijden door `?contactItemValue` te gebruiken in beide `UNION`-branches van de query. Met `SELECT ?contactItemValue` ga je dan een lange lijst krijgen.
Je kan dan ook `DISTINCT` gebruiken om te garanderen dat de lijst alleen unieke waarden bevat. Dan is de `new Set()` hieronder ook niet meer nodig.